packages/flutter/lib/src/cupertino/date_picker.dart

```diff
@@ -464,6 +464,21 @@ class _CupertinoDatePickerDateTimeState extends State<CupertinoDatePicker> {
     }
 
     previousHourIndex = selectedHour;
+
+    PaintingBinding.instance.systemFonts.addListener(_handleSystemFontsChange);
+  }
+
+  void _handleSystemFontsChange () {
+    estimatedColumnWidths.clear();
```

Comment:
Put this statement into the setState closure? (That's the state getting changed here, right?)